websites/S/Stremio/presence.ts

```diff
@@ -3,172 +3,404 @@ const presence = new Presence({
+						)?.textContent,
+						genre =
+							document
+								.querySelector(
+									appVersion === AppVersion.V4
+										? "ul.genre-select > li.selected"
+										: "div[class*='selectable-inputs-container'] > div:nth-child(3) > div"
+								)
+								?.textContent?.replace("Select genre", "") || null;
+
+					presenceData.buttons = [
+						{
+							label: "Browse",
+							url: href,
+						},
+					];
+					presenceData.details = `Discovering ${type ?? "content"}${
+						type === "series" ? "" : "s"
+					}`;
+					presenceData.state = `${category ?? "All"}${
+						genre ? ` | ${genre}` : ""
+					}`;
+
+					break;
+				}
+				case "library": {
+					const type = document.querySelector(
+						appVersion === AppVersion.V4
+							? "[class='ng-binding ng-scope selected']"
+							: "div[class*='selectable-inputs-container'] > div > div"
+					)?.textContent;
+
+					presenceData.details = "Library";
+					presenceData.state = type ?? "All";
+					presenceData.buttons = [
+						{
+							label: "View library",
+							url: href,
+						},
+					];
+					break;
+				}
+				case "calendar":
+					presenceData.details = "Calendar";
+					presenceData.buttons = [
+						{
+							label: "View calendar",
+							url: href,
+						},
+					];
+					break;
+				case "search":
+					presenceData.details = "Search";
+					break;
+				case "player": {
+					if (video === null) break;
+
+					presenceData.endTimestamp = video.endTimestamp;
+					delete presenceData.startTimestamp;
+
+					if (
+						(appVersion === AppVersion.V4
+							? document
+									.querySelector("#loading-logo")
+									.className.includes("flashing")
+							: !!document.querySelector(
+									"div[class*='buffering-loader-container']"
+							  )) ||
+						video.isPaused
+					) {
+						presenceData.smallImageKey = Assets.Pause;
+						presenceData.smallImageText = "Player is paused";
+						presenceData.state = "Paused";
+					} else {
+						presenceData.smallImageKey = Assets.Play;
+						presenceData.smallImageText = "Player is playing";
+						presenceData.state = "Watching";
+					}
+
+					let metaUrl: string, title: string;
+
+					if (appVersion === AppVersion.V4) {
+						title = document
+							.querySelector("head > title")
+							?.textContent?.replace("Stremio -", "")
+							?.trim();
+						metaUrl = href
+							.substring(0, href.lastIndexOf("/"))
+							.replace("player", "detail");
+						presenceData.largeImageKey =
+							document
+								.querySelector("#loading-logo")
+								?.getAttribute("data-image") ?? Assets.Logo;
+					} else {
+						const playerState = await _eval(
+							"core.transport.getState('player')"
+						);
+						if (playerState.metaItem.type.toLowerCase() === "ready") {
+							// eslint-disable-next-line prefer-destructuring
+							const content = playerState.metaItem.content;
+							// eslint-disable-next-line prefer-destructuring
```

Comment:
try to avoid fixing CI issues by adding `eslint-disable` directives, and prefer fixing the issue directly if possible.

Reply:
I know, and believe me I tried, but some of these rules are just unreasonably strict and actually make the code worse compared to when the warning is 'fixed'. A ton of other presences also disable some of the eslint rules, so I'm definitely not the only dev who experiences this.

Reply:
The highlighted llines would look like this when destructured:
```js
const {metaItem: {content}} = playerState;
{title} = playerState;
```

The former is less readable imo and the latter is invalid JS (`Declaration or statement expected. This '=' follows a block of statements, so if you intended to write a destructuring assignment, you might need to wrap the whole assignment in parentheses.ts
`).

If you absolutely prefer the former LOC for whatever personal reason let me know and I'll change it.

Reply:
Actually, never mind. I'll just change it. Eslint rules are there for a reason, this is your personal preference right now so I shouldn't even debate this. May be worth an issue though, I feel like in general these rules are way to strict.